enterprise/ha/src/main/java/org/neo4j/kernel/impl/ha/ClusterManager.java

```diff
@@ -123,8 +124,20 @@
         HighlyAvailableGraphDatabase repair() throws Throwable;
     }
 
+    public static IntFunction<String> constant( final String value )
+    {
+        return new IntFunction<String>()
+        {
+            @Override
+            public String apply( int ignored )
+            {
+                return value;
+            }
+        };
+    }
+
     private final File root;
-    private final Map<String,String> commonConfig;
+    private final Map<String,IntFunction<String>> commonConfig;
```

Comment:
I would say that limiting map value into IntFunction is too strict and brings an idea that for some reason common config should be always int dependent which is not true for all properties. Would make if wider to use plain Function instead.